mmdet/apis/test.py

```diff
@@ -2,25 +2,45 @@
 import pickle
 import shutil
 import tempfile
+from pathlib import Path
 
 import mmcv
 import torch
 import torch.distributed as dist
 from mmcv.runner import get_dist_info
 
+from mmdet.core import tensor2imgs
 
-def single_gpu_test(model, data_loader, show=False):
+
+def single_gpu_test(model, data_loader, show=False, images_out_dir=None):
     model.eval()
     results = []
     dataset = data_loader.dataset
     prog_bar = mmcv.ProgressBar(len(dataset))
     for i, data in enumerate(data_loader):
         with torch.no_grad():
-            result = model(return_loss=False, rescale=not show, **data)
+            result = model(
+                return_loss=False,
+                rescale=not (show or images_out_dir),
```

Comment:
We can set `rescale=True` to get the results on the original image scale, and then resize the image to `ori_shape` for visualization after Line36.

Reply:
Okey. That is a nice change. The current behaviour had caused me (and some others, I guess) problems in the past.